apps/dot_ext/views/authorization.py

```diff
@@ -295,38 +296,43 @@ def dispatch(self, request, uuid, *args, **kwargs):
 class TokenView(DotTokenView):
     @method_decorator(sensitive_post_parameters("password"))
     def post(self, request, *args, **kwargs):
-        """Subclass of oauth2_provider.views.base.TokenView
-        to support checking application active/DAG status and
-        modification of the access token to include access
-        grant expiry.
-
-        RETURN:
-            response: HTTPResponse containing serialized access
-            token response.
-        """
+        dag = None
+        dag_expiry = None
         try:
-            dag = validate_app_is_active(request)[1]
-            if dag is not None and dag.expiration_date is not None:
-                dag_expiry = strftime('%Y-%m-%d %H:%M:%S', dag.expiration_date.timetuple())
-            else:
-                dag_expiry = None
+            app = validate_app_is_active(request)
         except (InvalidClientError, InvalidGrantError) as error:
             return json_response_from_oauth2_error(error)
 
+        if app.data_access_type == "THIRTEEN_MONTH":
+            try:
+                dag = DataAccessGrant.objects.get(
+                    beneficiary=request.user.id,
+                    application=app
+                )
+                if dag is not None and dag.expiration_date is not None:
+                    dag_expiry = strftime('%Y-%m-%d %H:%M:%S', dag.expiration_date.timetuple())
+                else:
+                    dag_expiry = None
+            except DataAccessGrant.DoesNotExist:
+                future_dag_expiration = datetime.now() + timedelta(weeks=56)
+                dag_expiry = strftime('%Y-%m-%d %H:%M:%S', future_dag_expiration.timetuple())
+
         url, headers, body, status = self.create_token_response(request)
+
         if status == 200:
             body = json.loads(body)
             access_token = body.get("access_token")
+            if dag_expiry is None:
```

Comment:
We probably need a check for RESEARCH_STUDY apps here, I'm seeing research study apps still return a dag expiration date.